java/ql/src/Security/CWE/CWE-200/TempDirUtils.qll

```diff
@@ -0,0 +1,80 @@
+import java
+import semmle.code.java.dataflow.FlowSources
+
+/**
+ * A method that returns a `String` or `File` that has been tainted by `System.getProperty("java.io.tmpdir")`.
+ */
+abstract class MethodAccessSystemGetPropertyTempDirTainted extends MethodAccess { }
+
+/**
+ * Method access `System.getProperty("java.io.tmpdir")`.
+ */
+private class MethodAccessSystemGetPropertyTempDir extends MethodAccessSystemGetPropertyTempDirTainted,
+  MethodAccessSystemGetProperty {
+  MethodAccessSystemGetPropertyTempDir() {
+    this.hasCompileTimeConstantGetPropertyName("java.io.tmpdir")
+  }
+}
+
+/**
+ * A method call to the `org.apache.commons.io.FileUtils` methods `getTempDirectory` or `getTempDirectoryPath`.
+ */
+private class MethodAccessApacheFileUtilsTempDir extends MethodAccessSystemGetPropertyTempDirTainted {
+  MethodAccessApacheFileUtilsTempDir() {
+    exists(Method m |
+      m.getDeclaringType().hasQualifiedName("org.apache.commons.io", "FileUtils") and
+      m.hasName(["getTempDirectory", "getTempDirectoryPath"]) and
+      this.getMethod() = m
+    )
+  }
+}
+
+/**
+ * All `java.io.File::createTempFile` methods.
```

Comment:
QL is multivalued in general, so we're always dealing with sets of values. Thus, we prefer to specify qldoc in terms of the set elements rather than the set as that tends to be simpler and provides consistency.
```suggestion
 * A `java.io.File::createTempFile` method.
```
(And I assume overloads is the reason that there's presumably more than one - otherwise we'd write "The ...").